libyul/optimiser/UnusedFunctionsCommon.cpp

```diff
@@ -40,33 +40,33 @@ FunctionDefinition unusedFunctionsCommon::createLinkingFunction(
 	auto generateTypedName = [&](TypedName t)
 	{
 		return TypedName{
-			t.location,
+			t.debugData,
 			_nameDispenser.newName(t.name),
 			t.type
 		};
 	};
 
-	langutil::SourceLocation loc = _original.location;
+	std::shared_ptr<DebugData const> debugData = _original.debugData;
```

Comment:
```suggestion
	shared_ptr<DebugData const> debugData = _original.debugData;
```

And a useless local again.